### PR TITLE
add support for global request options

### PR DESCRIPTION
**Pull request by sf-csarov**

### CODE REVIEW

**Comment by vernak2539** [comment]
@sf-csarov looks good, could you please remove the `package-lock.json`? 

should be able to get it merged after that

File: lib/fuel-rest.js [hunk 1]
```diff
@@ -73,7 +74,15 @@ class FuelRest {
 		});
 	}
 	_processRequest(options, callback) {
-		this.AuthClient.getAccessToken(clone(options.auth))
+
+		const requestOptions = clone(options.auth) || {};
+
+        if (this.proxy) {
+            requestOptions.proxy = this.proxy;
+            requestOptions.rejectUnauthorized = false;
```
Comment (vernak2539) [inline lib/fuel-rest.js]: Is there a reason that this is passed by default? it's technically enabling a security hole if people don't look at this option in-depth

**Comment by sf-csarov** [comment]
I made rejectUnauthorized optional, and removed package-lock.json

**Comment by vernak2539** [comment]
We need to get the fuel auth PR merged first so we can increment the package version here when we are doing this PR. https://github.com/salesforce-marketingcloud/FuelSDK-Node-Auth/pull/71

**Comment by vernak2539** [comment]
`options.globalReqOptions` is used in both the auth and standard REST requests. Is this something that is expected? The fuel-auth module can be updated to `v3.1.0` as well

**Comment by sf-csarov** [comment]
Yes, why not. Basically, it changes the "request" package options. Both uses the same request package. Updated the version.

**Comment by vernak2539** [comment]
published as `fuel-rest@3.1.0`

**Pull request merged**